tests/ClientBuilderTest.php

```diff
@@ -210,6 +210,33 @@ public function optionsDataProvider()
             ['setErrorTypes', 0],
         ];
     }
+
+    /**
+     * @dataProvider compressionOptionDataProvider
+     */
+    public function testCompressionPlugin($enabled)
```

Comment:
Please typehint the return type (`void`) and the parameter. Also I would change the method name to `testGetClientTogglesCompressionPluginInHttpClient`